src/python/WMComponent/DBS3Buffer/MySQL/NewSubscription.py

```diff
@@ -40,15 +40,17 @@ def _createPhEDExSubBinds(self, datasetID, subscriptionInfo, custodialFlag):
 
         binds = []
         for site in sites:
-            binds.append({'id': datasetID,
-                          'site': site,
-                          'custodial': custodialFlag,
-                          'auto_approve': 1 if site in subscriptionInfo['AutoApproveSites'] else 0,
-                          'move': isMove,
-                          'priority': subscriptionInfo['Priority'],
-                          'phedex_group': phedex_group,
-                          'delete_blocks': delete_blocks,
-                          'dataset_lifetime': subscriptionInfo['DatasetLifetime']})
+            bind = {'id': datasetID,
+                    'site': site,
+                    'custodial': 0 if custodialFlag is None else custodialFlag,
```

Comment:
custodialFlag here can be passed as `-1` , it is still and integer but I doubt it is correct value, please validate it appropriately to fall to a specific range, e.g. be only 1 or 0.

Reply:
It looks like the `custodialFlag` is actually either True or False, as defined in this very same code. Unsure how was/is getting casted to integer though...